packages/blade/src/components/Input/OTPInput/OTPInput.tsx

```diff
@@ -214,6 +249,9 @@ const OTPInput = ({
     for (let index = 0; index < otpLength; index++) {
       const currentValue = inputValue ? otpToArray(inputValue)[index] || '' : otpValue[index] || '';
       const ref = React.createRef<HTMLInputElement>();
+      // if an inputValue is passed (controlled) and isMasked is set, inputType will always be password
+      const currentInputType =
+        inputValue && isMasked ? 'password' : isMasked ? inputType[index] : undefined;
```

Comment:
why undefined and not text?

Reply:
It was `undefined` before this too. I think we didn't set it as text explicitly since we weren't sure if our use-cases would be numeric or alphanumeric.